girder/molecules/molecules/utilities/molecules.py

```diff
@@ -32,10 +34,17 @@ def create_molecule(data_str, input_format, user, public, gen3d=True,
     using_2d_format = (input_format in openbabel_2d_formats)
     inchi_format = 'inchi'
 
-    if using_2d_format or input_format in openbabel_3d_formats:
+    if using_2d_format:
         inchi, inchikey = openbabel.to_inchi(data_str, input_format)
     else:
-        sdf_data = avogadro.convert_str(data_str, input_format, 'sdf')
+        # Let's make sure the bonds look reasonable
+        cjson = convert_3d_format_to_cjson(data_str, input_format)
+        if bonding_looks_suspicious(cjson):
+            tmp = openbabel.autodetect_bonds(cjson)
```

Comment:
Is `autodetect_bonds` computationally intensive? Should it be done asynchronously?

Reply:
I don't think it is, but I will give it a try on a big molecule to see how long it takes. I think it essentially loops through all atom-atom pairs and calculates the distance between them, and determines if the their distance is less than the sum of their covalent radii, or something like that.

Reply:
I gave it a try on a real molecule with 457 atoms (207 heavy atoms), and the `autodetect_bonds()` function there took about 0.04 seconds on my computer.

Reply:
Not an issue then :-)